backend/src/Parser/CanvasV2.fs

```diff
@@ -0,0 +1,179 @@
+module Parser.CanvasV2
+
+open FSharp.Compiler.Syntax
+
+open Prelude
+open Tablecloth
+
+module PT = LibExecution.ProgramTypes
+
+open Utils
+
+type CanvasModule =
+  { types : List<PT.UserType.T>
+    fns : List<PT.UserFunction.T>
+    dbs : List<PT.DB.T>
+    // TODO: consider breaking this down into httpHandlers, crons, workers, and repls
+    handlers : List<PT.Handler.Spec * PT.Expr>
+    exprs : List<PT.Expr> }
+
+let emptyModule = { types = []; fns = []; dbs = []; handlers = []; exprs = [] }
+
+
+/// Extracts the parts we care about from an F# attribute
+///
+/// Given `[<HttpHandler("method", "path")>]`, returns `("HttpHandler", ["method"; "path"])`
+/// wch is easier to work with than the AST presented normally
+let (|SimpleAttribute|_|) (attr : SynAttribute) =
+  let attrName =
+    match longIdentToList attr.TypeName.LongIdent with
+    | [ attrName ] -> attrName
+    | _ -> Exception.raiseInternal $"Unsupported attribute name" []
+
+  let rec parseAttrArgs (attr : SynExpr) : List<string> =
+    match attr with
+    | SynExpr.Paren (expr, _, _, _) -> parseAttrArgs expr
+
+    | SynExpr.Tuple (_, args, _, _) ->
+      args
+      |> List.map (fun arg ->
+        match arg with
+        | SynExpr.Const (SynConst.String (s, _, _), _) -> s
+        | _ ->
+          Exception.raiseInternal $"Couldn't parse attribute argument" [ "arg", arg ])
+
+    | _ ->
+      Exception.raiseInternal $"Couldn't parse attribute argument" [ "attr", attr ]
+
+  Some(attrName, parseAttrArgs attr.ArgExpr)
+
+/// Update a CanvasModule by parsing a single F# let binding
+/// Depending on the attribute present, this may add a user function, a handler, or a DB
+let parseLetBinding
+  availableTypes
+  (m : CanvasModule)
+  (letBinding : SynBinding)
+  : CanvasModule =
+  match letBinding with
+  | SynBinding (_, _, _, _, attrs, _, _, pat, _returnInfo, expr, _, _, _) ->
+    let expr = ProgramTypes.Expr.fromSynExpr availableTypes expr
+
+    let attrs = attrs |> List.collect (fun l -> l.Attributes)
+
+    let randomIds () : PT.Handler.ids =
+      { moduleID = gid (); nameID = gid (); modifierID = gid () }
+
+    match attrs with
+    | [] ->
+      let newFn = ProgramTypes.UserFunction.fromSynBinding availableTypes letBinding
+      { m with fns = newFn :: m.fns }
+
+    | [ attr ] ->
+      match attr with
+      | SimpleAttribute ("DB", [ name ]) ->
+        // let newDB = PT.DB(name, randomIds ())
+        // { m with dbs = newDB :: m.dbs }
+        Exception.raiseInternal
+          $"Not currently supporting DBs - probably makes sense to wait until they are based on a type rather than 'columns'"
```

Comment:
For now, using the v1 parser is probably fine - just an annotation on a type instead of a `let`.